### PR TITLE
Fix GH#25103: Infer <type> from <duration> in MusicXML import

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
Resolves: #25103

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
Doesn't work with tuplets lacking the `<type>` tag, not even when a `<note-type>` is present inside the `<time-modification>` tag

**Review by miiizen — APPROVED** [review]
This looks good to me, just needs a rebase! Thanks.

**Comment by Jojo-Schmitz** [comment]
Rebased, please recheck

**Comment by Jojo-Schmitz** [comment]
The uniti tests is fails on got added/modified in @miiizen's 28727f7a, with this PR here it looks like they'd need to get reverted to pass ?!?

File: src/importexport/musicxml/internal/musicxml/import/importmusicxmlpass2.cpp [hunk 1]
```diff
@@ -6030,18 +6030,14 @@ void MusicXmlParserPass2::divisions()
...
-
-    if (!mDura.isValid()) {
+    if (!restDuration.isValid() || !measureDuration.isValid()) {
         return false;
     }
 
-    return (type.empty() && dura == mDura)
-           || (type == u"whole" && dura == mDura);
+    return (type.empty() && restDuration == measureDuration)
+           || (type == u"whole" && restDuration == measureDuration && restDuration != Fraction(1, 1));
```
Comment (miiizen) [inline src/importexport/musicxml/internal/musicxml/import/importmusicxmlpass2.cpp:6040]:
```suggestion
           || (type == u"whole" && restDuration == measureDuration);
```
The unit tests are the desired outcome. This condition is causing the test failures - looks like it sneaked back in after the rebase.

**Pull request merged**